database/utils.py

```diff
@@ -106,9 +107,14 @@ def _get_value_from_archive(self, obj):
         return self.default_value
 
     def __get__(self, obj, objtype=None):
+        cached_value = getattr(obj, "__archive_field_cached_value", None)
```

Comment:
i think it'd be nice to keep all the cache-related stuff into `__get__()` and `__set__()` and out of the other functions
```
def __get__(self, obj, objtype=None):
    if getattr(obj, self.cached_value_name) is None:
        db_field = getattr(obj, self.db_field_name)
        if db_field is not None:
            value = ...
        else:
            value = self._get_value_from_archive(obj)
        setattr(obj, self.cached_value_name, value)
    return getattr(obj, self.cached_value_name)
```